medical_pathology_icd10/wizards/medical_pathology_import.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+import logging
+
+from io import BytesIO
+from xml.etree import ElementTree
+from urllib2 import urlopen
+from zipfile import ZipFile
+
+from odoo import api, models
+
+_logger = logging.getLogger(__name__)
+
+
+class MedicalPathologyImport(models.TransientModel):
+    """ It provides a model for importing ICD-10-CM codes to Odoo. """
+
+    _inherit = 'medical.pathology.import'
+
+    ICD10_URI = '/'.join((
+        'ftp://ftp.cdc.gov/pub/Health_Statistics/NCHS',
+        'Publications/ICD10CM/2017/ICD10CM_FY2017_Full_XML.zip',
+    ))
+    ICD10_FILE_NAME = 'icd10cm_tabular_2017.xml'
+    ICD10_CODE = 'medical_pathology.pathology_code_05'
+    ICD10_XML_PREFIX = 'medical_pathology_icd10_test'
+
+    @api.model
+    def _get_importer_types(self):
+        res = super(MedicalPathologyImport, self)._get_importer_types()
+        return res + [('icd10', 'ICD-10-CM')]
+
+    @api.multi
```

Comment:
The decorator isn't needed here because this method is actually getting called from a proper onchange defined in ``medical_pathology_import``:

```
    @api.multi
    @api.onchange('importer_type')
    def _onchange_importer_type(self):
        """ It calls gets default data by importer type.

        To plug into this interface, importers should implement method
        ``_onchange_importer_type_*``, such as
        ``_onchange_importer_type_icd10``.
        """
        try:
            method = getattr(
                self, '_onchange_importer_type_%s' % self.importer_type,
            )
            method()
        except AttributeError:
            self.zip_uri = False
            self.file_name = False
```

That also takes care of the conditional logic.